c10/core/MemoryFormat.h

```diff
@@ -127,7 +127,7 @@ inline bool is_channels_last_strides_2d_s4(
     if (sizes[d] == 0) {
       return false;
     }
-    if (strides[d] < min) {
+    if (strides[d] < min && !(d == 0 && sizes[d] == 1)) {
```

Comment:
Hmm, can you please explain why this is necessary? Also, if this check is always for sizes[0], wouldn't it be faster to replace it as suggested (so that two checks can be done in parallel)
```suggestion
    if (strides[d] < min && !(d == 0 && sizes[0] == 1)) {
```

Reply:
As described in https://github.com/pytorch/pytorch/issues/78611, reshape of channels last tensor may get wrong stride on the dim where the size is 1. So we can skip the check of `strides[d] < min` when `d == 0` (size of dim 0 is 1 is the real scenario or we can also skip all the checks when sizes[d] == 1 since when size == 1 the stride has no other effect).

I will modify it with `strides[d] < min && !(d == 0 && sizes[0] == 1)` or `strides[d] < min && sizes[d] !=1`

Reply:
If skip all the checks when sizes[d] == 1  i.e., `strides[d] < min && sizes[d] !=1`. We should also fix corner cases.
For example : on aten/src/ATen/test/memory_format_test.cpp.
```
  t = at::rand({1, 4, 4, 1});
  t.transpose_(1, 3);
  EXPECT_TRUE(t.suggest_memory_format() != at::MemoryFormat::ChannelsLast);
```
The shape of transposed t: [1 4, 1, 4]. The stride of  transposed t: [16, 1, 4, 4].
Transposed t will be considered channels last.